andela/helios-ah-backend
# #161779785, #161779787 Add like and rating functionality

#### What does this PR do?
This PR allows a user to like/unlike an article and likewise, rate an article

#### Description of Task to be completed?
* Implements the like controller
* Implements the rating Controller
* Creates a feedback table having columns `id`, 
    `articleId`, `userId`, `isLiked` and `rating`
* Implement middleware to check if an article has
   been initially liked/rated by the user before adding 
   a new set of data to the feedback table
* Add test cases for the like controller
* Add test cases for the rating Controller

#### How should this be manually tested?
* clone the repository  ` https://github.com/andela/helios-ah-backend.git `
* cd to the  ` helios-ah-backend ` folder
* check out the `ft-user-like-and-rate-article-161779747` branch
* run `npm i` on the bash terminal
* run `npm run test` on the terminal

#### What are the relevant pivotal tracker stories?
#161779785, #161779787

#### Any background context you want to add?
N/A

#### Screenshots
<img width="862" alt="screen shot 2018-12-14 at 4 40 27 am" src="https://user-images.githubusercontent.com/25525765/49982319-5270d300-ff5c-11e8-8507-c0f47ee4c68a.png">
